### PR TITLE
Chore: Add plugins dskit module

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

This is a naive attempt to move plugin initialization to a dskit service.

**Why do we need this feature?**

This is needed to disable plugin initialization when running other targets in microservice deployment mode.

**Who is this feature for?**

@grafana/grafana-app-platform-squad 

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/manager/store/store_test.go [hunk 1]
```diff
@@ -42,7 +42,8 @@ func TestStore_ProvideService(t *testing.T) {
 			}
 		}}
 
-		_, err := ProvideService(fakes.NewFakePluginRegistry(), srcs, l)
+		s, err := ProvideService(fakes.NewFakePluginRegistry(), srcs, l)
+		s.starting(context.Background())
```
Comment (DanCech) [inline pkg/plugins/manager/store/store_test.go:46]: Should we be calling StartAsync and deferring StopAsync here?

File: pkg/api/plugin_resource_test.go [hunk 2]
```diff
@@ -77,6 +78,10 @@ func TestCallResource(t *testing.T) {
 	ps, err := store.ProvideService(reg, srcs, l)
 	require.NoError(t, err)
 
+	_ = ps.StartAsync(context.Background())
```
[the remark this answers is not in the document]
Comment (toddtreece) [reply pkg/api/plugin_resource_test.go:81]: i think this was to make lint happy, but i will actually check these errors

**Review by wbrowne — COMMENTED** [review]
Unfortunately this change breaks a dependency that was created with the introduction of the secrets manager plugin 😞 - see [here](https://github.com/grafana/grafana/blob/main/pkg/services/secrets/kvstore/kvstore.go#L34). AFAIK this is the only place in Grafana where this is expected.

Two ways around this:
- Migrate the secrets KV store to dskit
- Import the concrete plugin store and start/await the service to run (hacky, but I _think_ it should work)

```go
 	// ...
	"github.com/grafana/grafana/pkg/plugins/manager/store"
)

func ProvideService(
	pluginsManager *store.Service,
        // ...
       )
	var logger = log.New("secrets.kvstore")
	var store SecretsKVStore
	ctx := context.Background()
	store = NewSQLSecretsKVStore(sqlStore, secretsService, logger)

	_ = pluginsManager.StartAsync(ctx)
	_ = pluginsManager.AwaitRunning(ctx)
	// ...
```

**Comment by toddtreece** [comment]
> Migrate the secrets KV store to dskit

@wbrowne thanks for catching this. i'll handle this in a separate PR and get that merged before continuing with this one

File: pkg/plugins/manager/store/store_test.go (hunk 1 above)
Comment (toddtreece) [reply pkg/plugins/manager/store/store_test.go]: updated, thanks!

File: pkg/api/plugin_resource_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (toddtreece) [reply pkg/api/plugin_resource_test.go:81]: ✅ updated

**Pull request closed without merging**